channel/funder.go

```diff
@@ -325,27 +322,41 @@ func (f *Funder) depositedSub(ctx context.Context, contract *bind.BoundContract,
 	return sub, errors.WithMessage(err, "subscribing to deposited event")
 }
 
-// waitForFundingConfirmation waits for the confirmation events on the blockchain that
-// both we and all peers successfully funded the channel for the specified asset
-// according to the funding agreement.
-func (f *Funder) waitForFundingConfirmation(ctx context.Context, request channel.FundingReq, asset assetHolder, fundingIDs [][32]byte) error {
+func (f *Funder) subscribeDeposited(ctx context.Context, contract *bind.BoundContract, fundingIDs ...[32]byte) (chan *subscription.Event, *subscription.ResistantEventSub, chan error, error) {
 	deposited := make(chan *subscription.Event)
 	subErr := make(chan error, 1)
 	// Subscribe to events.
-	sub, err := f.depositedSub(ctx, asset.contract, fundingIDs...)
+	sub, err := f.depositedSub(ctx, contract, fundingIDs...)
 	if err != nil {
-		return errors.WithMessage(err, "subscribing to deposited event")
+		return nil, nil, nil, errors.WithMessage(err, "subscribing to deposited event")
 	}
-	defer sub.Close()
 	// Read from the sub.
 	go func() {
 		subErr <- sub.Read(ctx, deposited)
 	}()
+	return deposited, sub, subErr, nil
+}
+
+// waitForFundingConfirmation waits for the confirmation events on the blockchain that
+// both we and all peers successfully funded the channel for the specified asset
+// according to the funding agreement.
+func (f *Funder) waitForFundingConfirmation(ctx context.Context, request channel.FundingReq, asset assetHolder, fundingIDs [][32]byte) error {
+	// Subscribe to events.
+	deposited, sub, subErr, err := f.subscribeDeposited(ctx, asset.contract, fundingIDs...)
+	if err != nil {
+		return errors.WithMessage(err, "subscribing to deposited event")
+	}
+	defer sub.Close()
 
 	// The allocation that all participants agreed on.
 	agreement := request.Agreement.Clone()[asset.assetIndex]
-	// Count how many zero balance funding requests are there
-	N := len(request.Params.Parts) - countZeroBalances(agreement)
+	// Count how often we don't expect an event because we have a zero balance or
+	// the asset is on a different ledger.
```

Comment:
I think it's debatable which order is better :D

Reply:
Yes, but I just meant that the comment is consistent with the code because you're first checking the chains and then the zero balances. (I was first a little confused because I thought the method countDifferentLedger does both).